sigma/backends/netwitness/netwitness.py

```diff
@@ -0,0 +1,384 @@
+
+from sigma.conditions import (
+    ConditionAND,
+    ConditionFieldEqualsValueExpression,
+    ConditionItem,
+    ConditionNOT,
+    ConditionOR,
+    ConditionValueExpression,
+)
+from sigma.conversion.base import TextQueryBackend
+from sigma.conversion.deferred import DeferredQueryExpression
+from sigma.conversion.state import ConversionState
+from sigma.exceptions import SigmaConversionError
+from sigma.types import (
+    SigmaCompareExpression,
+    SigmaExpansion,
+    SigmaNumber,
+    SigmaRegularExpressionFlag,
+    SigmaString,
+    SpecialChars,
+)
+
+
+class NetWitnessBackend(TextQueryBackend):
+    """NetWitness backend."""
+
+    name: ClassVar[str] = "netwitness backend"
+    formats: ClassVar[Dict[str, str]] = {
+        "default": "Plain netwitness queries",
+    }
+    # TODO: does the backend requires that a processing pipeline is provided? This information can be used by user
+    # interface programs like Sigma CLI to warn users about inappropriate usage of the backend.
+    requires_pipeline: ClassVar[bool] = False
+
+    # Operator precedence: tuple of Condition{AND,OR,NOT} in order of precedence.
+    # The backend generates grouping if required
+    precedence: ClassVar[Tuple[ConditionItem, ConditionItem, ConditionItem]] = (ConditionNOT, ConditionAND, ConditionOR)
+    group_expression: ClassVar[Optional[str]] = (
+        "({expr})"  # Expression for precedence override grouping as format string with {expr} placeholder
+    )
+    parenthesize: bool = True
+
+    # Generated query tokens
+    token_separator: str = " "  # separator inserted between all boolean operators
+    or_token: ClassVar[Optional[str]] = "||"
+    and_token: ClassVar[Optional[str]] = "&&"
+    not_token: ClassVar[Optional[str]] = "NOT"
+    eq_token: ClassVar[Optional[str]] = " = "  # Token inserted between field and value (without separator)
+
+    # String output
+    # Fields
+    # Quoting
+
+    # Character used to quote field characters if field_quote_pattern matches (or not, depending on
+    # field_quote_pattern_negation). No field name quoting is done if not set.
+    field_quote: ClassVar[Optional[str]] = None
+    # Quote field names if this pattern (doesn't) matches, depending on field_quote_pattern_negation.
+    # Field name is always quoted if pattern is not set.
+    field_quote_pattern: ClassVar[Optional[Pattern[str]]] = re.compile("^\\w+$")
+    # Negate field_quote_pattern result. Field name is quoted if pattern doesn't matches if set to True (default).
+    field_quote_pattern_negation: ClassVar[bool] = True
+
+    # Values
+    str_quote: ClassVar[str] = "'"  # string quoting character (added as escaping character)
+    escape_char: ClassVar[Optional[str]] = "\\"  # Escaping character for special characrers inside string
+    wildcard_multi: ClassVar[Optional[str]] = ""  # Character used as multi-character wildcard
+    wildcard_single: ClassVar[Optional[str]] = ""  # Character used as single-character wildcard
```

Comment:
You probably want to remove this so it leads to an error if something like it is used? This applies to other missing features as well i guess.

Reply:
These are needed as empty strings to convert contains, startswith, endswith into list expressions